src/main/java/com/sumologic/client/Credentials.java

```diff
@@ -7,37 +7,62 @@
  * @author Sebastian Mies
  */
 public class Credentials {
-    private String email;
-    private String password;
+    private String accessId;
+    private String accessKey;
 
     /**
-     * Constructs the credentials using email and password.
+     * Constructs the credentials using accessId and accessKey.
      *
-     * @param email Your email
-     * @param password Your password
+     * @param accessId Your access id
+     * @param accessKey Your access key
      */
-    public Credentials(String email, String password) {
-        this.email = email;
-        this.password = password;
+    public Credentials(String accessId, String accessKey) {
```

Comment:
`SumoLogicClient` will have identical signature too, and that one is the main one used by users to initialize the client. If we need to create a new one, we need change that one too.

I see that this approach will have benefit that if user is upgrading to a new client version, they will get a compilation warning. This works for a deprecation that we have enough soak time to let the user discover and change their code. But I think this is not our case and we are not breaking any user. We don't have time to let user discover it and fix (typically we need keep deprecated class in several releases). Instead we actively track and communicate EVERY user that using email to access api on a weekly basis and ask them to replace it with access key. 

Also I'm worried about making a new class here since a lot of user already changed their code to use access key with the old class and they think they are good. If in the future, when we remove the old class, those users are required to change their code again, which is not ideal. It will also need an extra round communication effort (and it's much harder since not like today, we actually have no way to tell which user doing access key using deprecated class).